subprojects/language-native/src/integTest/groovy/org/gradle/language/swift/tasks/UnexportMainSymbolIntegrationTest.groovy

```diff
@@ -35,10 +37,31 @@ class UnexportMainSymbolIntegrationTest extends AbstractInstalledToolChainIntegr
         """
     }
 
+    @Issue("https://github.com/gradle/gradle-native/issues/304")
+    def "clean build works"() {
+        writeMainSwift()
+
+        expect:
+        succeeds("clean", "unexport", "assemble")
```

Comment:
Why do you need to call `unexport`? Should that task be implementation detail and doesn't expect a user to call that task directly. I was able to get the same behavior with `succeeds("clean", "test")`. I don't know if that has the same intent as this test. What do you think?

Reply:
This is a simplified version of what the test plugin does (the build script directly creates the `UnexportMainSymbol` task).

We're calling `unexport` because there's nothing actually using the output of `unexport`.  We're just looking at the generated object file.  There are some tests in `SwiftXCTestIngrationTest` that uses the test plugin and `UnexportMainSymbol` task.

The calls to `assemble` are actually misleading in the other tests.  There used to be something more complicated here that checked that the final executable had a valid `_main` symbol.  I went ahead and removed them.